src/misc.rs

```diff
@@ -0,0 +1,11 @@
+use esp_idf_sys::*;
+
+pub fn get_default_efuse_mac() -> Result<[u8; 6], EspError> {
+    let mut mac = [0; 6];
+    unsafe { esp!(esp_efuse_mac_get_default(mac.as_mut_ptr()))? }
+    Ok(mac)
+}
+
+pub fn restart() {
```

Comment:
I think this is overall useful and can stay. Ideally, we should have a trait for it in `embedded-svc`, but this can come later (at the expense of API breakage).